pkg/functionconfig/reader.go

```diff
@@ -36,16 +38,70 @@ func NewReader(parentLogger logger.Logger) (*Reader, error) {
 	}, nil
 }
 
+// Merges codeEntry config with base function config.
+// Base config populated values won't get override by codeEntry values.
+// The only exception is that there will be a union of the environment variables, with precedence to base config.
```

Comment:
Suggest to flip the terminology - you're (for the most part) merging the user config on top of the  `codeEntryConfig`, so `codeEntryConfig` is the base for the merge